### PR TITLE
174915493 student sort

### PR DESCRIPTION
**Pull request by mklewandowski**
This PR syncs the state of the two student sort `CustomSelect` component instances within the app.  Unlike other occurrences of the `CustomSelect` where the selected item is managed by the component's local state, if the current sort method is changed in one `CustomSelect`, it must also be changed in the other instance (this is similar to what we needed to do with the 3(!) anonymize toggles in the app - they all have to have a synchronized state).  To do this, we obtain the current student sort method from the Redux component and pass it to the `CustomSelect`.  This allows the `CustomSelect` to use or override its internal component state management based on the absence or presence of an optional prop.

There is also some minor refactoring of component props lists.  As the props lists grow on these components (a side effect of an early choice to grab the Redux state in the main app component and then pass it down as needed to child components), I'm finding it helpful to have them in alphabetical order.

### CODE REVIEW

File: js/components/portal-dashboard/custom-select.tsx
Comment (kswenson) [inline js/components/portal-dashboard/custom-select.tsx:15]: As suggested below, I think the naming could be clearer here. If I were just naming this prop I'd probably name it `selectedItem` for consistency with `onSelectItem`, but that's not ideal because it's not really an item. If we think of this as a simple control like any other, e.g. a standard `<select>`, then the props should be named `value` and `onChange`. Then perhaps the fields of `SelectItem` should be named `value` and `label`. More broadly, I think there was a notion that `CustomSelect` was a component that could be standardized for use across multiple CC applications, but instead each usage has been customized in unique ways so that much of the benefit of consistency has been lost. Ideally, any changes to this component should be made with an eye toward general applicability rather than local expediency. I'm not sure how much of this is worth addressing now, but it's something to think about.

File: js/components/portal-dashboard/custom-select.tsx
Comment (kswenson) [inline js/components/portal-dashboard/custom-select.tsx:59]: I find the naming here somewhat confusing. `selectState` is a prop not state. `selectedItem` appears to be an action not an item. The `currentItem` is presumably also the selected item.

File: js/components/portal-dashboard/custom-select.tsx
Comment (kswenson) [inline js/components/portal-dashboard/custom-select.tsx:74]:
```suggestion
    const selectedItem = selectState || this.state.selected;
```

File: js/components/portal-dashboard/custom-select.tsx
Comment (kswenson) [inline js/components/portal-dashboard/custom-select.tsx:58]:
```suggestion
    const selectedItem = selectState || this.state.selected;
```

**Review by kswenson — APPROVED** [review]
👍 I have some misgivings about the direction that these changes are taking the `CustomSelect` component but I don't know that there's an easy way to address them so I leave it to you to decide to what extent changes should be made now rather than put off to later.

File: js/components/portal-dashboard/custom-select.tsx
Comment (mklewandowski) [reply js/components/portal-dashboard/custom-select.tsx:34]: I think you're right.  We were lucking out that the initial state was correct., but I believe it needs to be derived from either of the props.

File: js/components/portal-dashboard/custom-select.tsx
Comment (mklewandowski) [reply js/components/portal-dashboard/custom-select.tsx:15]: I like these changes.  In making them, I would also vote to turn the state variable `selected` also into `value` so we have consistency with the prop.  I also agree on attempting to make things like the `CustomSelect` and the `ToggleControl` generic and reusable in other apps.  That might be a fun side-project to add them to the CC components library that others were working on:
https://github.com/concord-consortium/concord-react-components

Comment (mklewandowski) [reply js/components/portal-dashboard/custom-select.tsx:15]: And in line with your naming ideas, I think we can turn `handleListClick` into `handleChange` for consistency.

File: js/components/portal-dashboard/custom-select.tsx
Comment (mklewandowski) [reply js/components/portal-dashboard/custom-select.tsx]: Fixed.

File: js/components/portal-dashboard/custom-select.tsx
Comment (mklewandowski) [reply js/components/portal-dashboard/custom-select.tsx]: I made the above changes.

File: js/components/portal-dashboard/custom-select.tsx
Comment (mklewandowski) [reply js/components/portal-dashboard/custom-select.tsx]: Fixed.

File: js/components/portal-dashboard/custom-select.tsx
Comment (mklewandowski) [reply js/components/portal-dashboard/custom-select.tsx]: Fixed.

**Review by eireland — APPROVED** [review]
Looks good to me

**Review by mklewandowski — COMMENTED** [review]
> Looks good to me

I'll wait until @kswenson has a chance to look at my select changes before merging.

File: js/components/portal-dashboard/custom-select.tsx
Comment (kswenson) [inline js/components/portal-dashboard/custom-select.tsx:66]:
```suggestion
        <div className={css.current}>{currentItem?.label}</div>
```

**Review by kswenson — APPROVED** [review]
👍 Looks good

File: js/components/portal-dashboard/custom-select.tsx
Comment (mklewandowski) [reply js/components/portal-dashboard/custom-select.tsx]: Fixed.

**Pull request merged**
